GitUI/UserControls/RevisionGrid.cs

```diff
@@ -991,11 +991,26 @@ private void RevisionsSelectionChanged(object sender, EventArgs e)
 
             var selectedRevisions = GetSelectedRevisions();
             var firstSelectedRevision = selectedRevisions.FirstOrDefault();
+            var secondSelectedRevision = selectedRevisions.Skip(1).FirstOrDefault();
             if (selectedRevisions.Count == 1 && firstSelectedRevision != null)
             {
                 _navigationHistory.Push(firstSelectedRevision.Guid);
+                compareToWorkingDirectoryMenuItem.Enabled = firstSelectedRevision.ObjectId != ObjectId.UnstagedId;
             }
 
+            // Bug in git-for-windows: Comparing working directory to any branch, fails, due to -R
+            // I.e., git difftool --gui --no-prompt --dir-diff -R HEAD fails, but
+            // git difftool --gui --no-prompt --dir-diff HEAD succeeds
+            // Thus, we disable comparing "from" working directory.
+            var enabledDiffOnWorkingDirectory = !(AppSettings.UseDifftoolDirDiff && firstSelectedRevision?.Guid == GitRevision.UnstagedGuid);
+
+            compareSelectedCommitsMenuItem.Enabled = selectedRevisions.Count == 2 && firstSelectedRevision != null && secondSelectedRevision != null && enabledDiffOnWorkingDirectory;
```

Comment:
Most behavior (at least in this menu) is allowing more than on revisions when selecting (the last two clicked are used). This changes this slightly

Reply:
I believe I've changed this to work as you mention.